File: identityscanner_v2/src/main/java/com/digitify/identityscanner/base/BaseActivity.java
package com.digitify.identityscanner.base;

import android.os.Bundle;
import android.view.WindowManager;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.LifecycleOwner;

import com.digitify.identityscanner.interfaces.IBase;

public abstract class BaseActivity extends AppCompatActivity implements IBase.View, LifecycleOwner {

    public void showToast(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_BLUR_BEHIND,
                WindowManager.LayoutParams.FLAG_BLUR_BEHIND);
    }
}

File: app/src/main/java/co/yap/app/di/BaseActivity.kt
package co.yap.app.di

import android.os.Bundle
import co.yap.app.di.component.DaggerActivityComponent
import co.yap.app.di.module.ActivityModule
import co.yap.yapcore.BaseActivity
import co.yap.yapcore.IBase

abstract class BaseActivity<V: IBase.ViewModel<*>> : BaseActivity<V>() {
    lateinit var daggerComponent: DaggerActivityComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        performInjection()
    }

    private fun performInjection() {
        daggerComponent = DaggerActivityComponent.builder()
            .activityModule(ActivityModule(this))
            .build() as DaggerActivityComponent
    }
}